Removes additional columns in test_owners.csv

calebamiles opened the pull request:
**What this PR does / why we need it**:

fixes a huge collection of typos with the number of columns in the CSV file that probably has broken the auto assign bot

**Special notes for your reviewer**:

None

**Release note**:

`NONE`

Commit:
Cleans up test/test_owners.csv

- sorts all e2e tests (with sort (GNU coreutils) 8.25)
- moves all k8s.io/* tests to the end
- removes duplicated tests (with uniq (GNU coreutils) 8.25)

calebamiles commented:
Sorry for breaking the bot!

**cc: @ethernetdan, @pwittrock, @kubernetes/test-infra-reviewers **

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/41660)
<!-- Reviewable:end -->

rmmh commented:
/lgtm

liggitt commented:
looks like it also adds about 200 lines... that intentional?

ethernetdan commented on test/test_owners.csv:
```diff
@@ -1,535 +1,738 @@
+HA-master survive addition/removal replicas different zones,derekwaynecarr,0,api-machinery
+HA-master survive addition/removal replicas multizone workers,rkouj,0,api-machinery
+HA-master survive addition/removal replicas same zone,derekwaynecarr,0,api-machinery
+AppArmor should enforce an AppArmor profile,derekwaynecarr,0,node
```
@calebamiles Looks like the file got repeated here. Could it be an issue with the [`hack/update_owners.py`](https://github.com/kubernetes/kubernetes/blob/master/hack/update_owners.py) script?

calebamiles commented:
- @ethernetdan, should be fixed
- @liggitt, nope wasn't intentional; should be fixed now

Commit:
Update tests listed in test/test_owners.py

- Adds SIG Storage ownership of Projected* tests
  - discussion: https://github.com/kubernetes/kubernetes/issues/19762
  - proposal: https://github.com/kubernetes/kubernetes/pull/35313

ethernetdan commented:
@k8s-bot gci gke e2e test this
@k8s-bot cvm gke e2e test this
@k8s-bot cvm gce e2e test this

ethernetdan commented:
@k8s-bot non-cri e2e test this
@k8s-bot gce etcd3 e2e test this
@k8s-bot gci gce e2e test this

ethernetdan commented:
Related to kubernetes/test-infra#1935

ethernetdan commented:
@rmmh @liggitt does this look good now?

spxtr commented:
/approve

rmmh commented:
/lgtm
